contracts/Vault.sol

```diff
@@ -663,9 +723,9 @@ contract Vault is
     function _transferAndCheckUnderlying(address _from, uint256 _amount)
         internal
     {
-        uint256 balanceBefore = totalUnderlying();
+        uint256 balanceBefore = underlying.balanceOf(address(this));
```

Comment:
I'm lost, what's this change about?

Reply:
I guess it's an equivalent way of checking the amount, but without doing unnecessary cross-contract calls to the strategy

Reply:
yes